contracts/finality/src/exec/finality.rs

```diff
@@ -63,7 +77,64 @@ pub fn handle_public_randomness_commit(
     Ok(Response::new().add_event(event))
 }
 
-// Copied from contracts/btc-staking/src/finality.rs
+/// Validates all parameters for public randomness commitment
+fn validate_pub_rand_commit(
+    env: &Env,
+    fp_btc_pk_hex: &str,
+    start_height: u64,
+    num_pub_rand: u64,
+    commitment: &[u8],
+    signature: &[u8],
+    min_pub_rand: u64,
+) -> Result<(), ContractError> {
+    // Check if FP BTC PubKey is empty
+    if fp_btc_pk_hex.is_empty() {
+        return Err(ContractError::EmptyFpBtcPubKey);
+    }
+
+    // Check if commitment is exactly 32 bytes
+    if commitment.len() != EXPECTED_COMMITMENT_LENGTH_BYTES {
+        return Err(ContractError::InvalidCommitmentLength {
+            expected: EXPECTED_COMMITMENT_LENGTH_BYTES,
+            actual: commitment.len(),
+        });
+    }
+
+    // Check for overflow when doing (StartHeight + NumPubRand)
+    // To avoid public randomness reset
+    let end_height = start_height.saturating_add(num_pub_rand);
+    if start_height >= end_height {
+        return Err(ContractError::OverflowInBlockHeight {
+            start_height,
+            end_height,
+        });
+    }
+
+    // Check if signature is empty
+    if signature.is_empty() {
+        return Err(ContractError::EmptySignature);
+    }
+
+    // Check if start height is too far into the future
+    let current_height = env.block.height;
+    if start_height >= current_height + MAX_PUB_RAND_COMMIT_OFFSET {
+        return Err(ContractError::InvalidPubRandHeight(
+            start_height,
+            current_height + MAX_PUB_RAND_COMMIT_OFFSET - 1,
+        ));
+    }
+
+    // Validate minimum public randomness requirement
+    if num_pub_rand < min_pub_rand {
+        return Err(ContractError::TooFewPubRand {
+            given: num_pub_rand,
+            required: min_pub_rand,
+        });
+    }
```

Comment:
https://github.com/babylonlabs-io/rollup-bsn-contracts/blob/5fe461327296d0cbcfc9a3ea9b8a245ad28e44b4/contracts/finality/src/state/public_randomness.rs#L169-L172 can be removed due to this change